Drive/rover_drive_system.hpp

```diff
@@ -34,63 +44,130 @@ class RoverDriveSystem
 
   void Enable(bool enable = true)
   {
-    rover_is_operational_ = enable;
+    mission_control_data_.is_operational = true;
     left_wheel_.Enable(enable);
     right_wheel_.Enable(enable);
     back_wheel_.Enable(enable);
-    SetMode('S');
+    SetMode();
   }
 
-  /// Main function for handling all the rover drive system functionality. Might
-  /// not need to take in params since maybe GetMissionControlData() will be
-  /// responsible for grabbing commands and saving them within the
-  /// RoverDriveSystem class.
-  void Move(char mode,
-            units::angle::degree_t rotation_angle,
-            units::angular_velocity::revolutions_per_minute_t speed)
+  // Retrieves /drive Mission Control commands.
+  /// @return returns true if connection is established from mission control
+  bool GetMissionControlData()
   {
-    if (rover_is_operational_)
+    // GET /drive JSON value ?
+    if (true)  // change 'true' to something for localhost connection found
+    {
+      char response[] = {};
+      ParseMissionControlData(response);
+      Move(mission_control_data_.drive_mode,
+           mission_control_data_.rotation_angle, mission_control_data_.speed);
+      return true;
```

Comment:
This probably shouldn't be within the GetData section of the code. Because its doing more than simply Getting data. It is now parsing it and also acting upon it.

Reply:
okay done